### PR TITLE
Update Spark Converter API section in README.rst

### PR DESCRIPTION
**Pull request by liangz1**
The "Spark Converter API" section now contains minimalist examples of Spark -> Tensorflow and Spark -> PyTorch.
I also edited the section "PySpark and SQL" to make it better connected with the preceding section.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -225,19 +225,24 @@ The minimalist example below assumes the definition of a ``Net`` class and
                                 transform_spec=transform), batch_size=1000) as test_loader:
         test(model, device, test_loader)
 
-Spark converter API
+Spark Converter API
```
Comment (WeichenXu123) [inline README.rst]: Spark Dataset Converter API

File: README.rst [hunk 2]
```diff
@@ -248,20 +253,44 @@ To simplify data conversion from Spark to TensorFlow, you can use Spark converte
...
+column followed by a label column.
+
+.. code-block:: python
+
+    from petastorm.spark import SparkDatasetConverter, make_spark_converter
+
+    spark.conf.set(SparkDatasetConverter.PARENT_CACHE_DIR_URL_CONF, 'hdfs:/...')
+
+    df1 = ... # `df1` is a spark dataframe
+    model = Net()
```
Comment (WeichenXu123) [inline README.rst]: What's the Net ? An empty model ?

File: README.rst [hunk 3]
```diff
@@ -248,20 +253,44 @@ To simplify data conversion from Spark to TensorFlow, you can use Spark converte
...
+
+    converter1 = make_spark_converter(df1)
+
+    with converter1.make_torch_dataloader() as dataloader:
+        train(model, dataloader, ...)
+
+    with converter1.make_torch_dataloader() as dataloader:
+        test(model, dataloader, ...)
```
Comment (WeichenXu123) [inline README.rst]: We cannot train and test one model on the same dataset?
Split the dataset and one for training and another for test...

File: README.rst (hunk 2 above)
Comment (liangz1) [reply README.rst]: I will mention in the description: "The minimalist example below assumes the definition of a ``Net`` class and
``train`` and ``test`` functions, included in ``pytorch_example``, ..."

**Comment by liangz1** [comment]
@selitvin Please feel free to review this PR, thanks!

File: README.rst [hunk 4]
```diff
@@ -248,20 +253,47 @@ To simplify data conversion from Spark to TensorFlow, you can use Spark converte
...
 
     # delete the cached files of the dataframe.
     converter1.delete()
 
+The minimalist example below assumes the definition of a ``Net`` class and
+``train`` and ``test`` functions, included in
+`pytorch_example.py <https://github.com/uber/petastorm/blob/master/examples/mnist/pytorch_example.py>`_,
+and a Spark DataFrame containing a feature column followed by a label column.
+
+.. code-block:: python
```
Comment (mengxr) [inline README.rst]: We should keep keras and torch examples the same. Do not expect a keras user will read pytorch example code. I think the torch one should just mirror the keras example to keep README simple.

File: README.rst [hunk 5]
```diff
@@ -248,20 +253,47 @@ To simplify data conversion from Spark to TensorFlow, you can use Spark converte
...
+
+    from petastorm.spark import SparkDatasetConverter, make_spark_converter
```
Comment (mengxr) [inline README.rst]: SparkDatasetConverter is not needed. Please also update the keras example.

File: README.rst [hunk 6]
```diff
@@ -248,20 +253,47 @@ To simplify data conversion from Spark to TensorFlow, you can use Spark converte
     # make a tensorflow dataset from `converter1
     with converter1.make_tf_dataset() as dataset:
```
Comment (mengxr) [inline README.rst]: There is only one DF and one converter in this example. Let's just call them df and converter.

File: README.rst [hunk 7]
```diff
@@ -248,20 +253,47 @@ To simplify data conversion from Spark to TensorFlow, you can use Spark converte
...
+        train(model, dataloader, ...)
+
+    with converter_test.make_torch_dataloader() as dataloader:
+        test(model, dataloader, ...)
+
+    converter_train.delete()
+    converter_test.delete()
+
 
 PySpark and SQL
```
Comment (mengxr) [inline README.rst]: Could you change the section title to "Analyzing petastorm datasets using PySpark and SQL"? cc: @selitvin

**Comment by mengxr** [comment]
@selitvin We are waiting for Spark 3.0 RC1 release to add an example that automatically converters MLlib vectors into dense arrays. Will do as a follow-up task.

File: README.rst (hunk 5 above)
Comment (liangz1) [reply README.rst]: It is used in L274 to refer to the `PARENT_CACHE_DIR_URL_CONF`.

**Pull request merged**